src/DssSpellCollateral.sol

```diff
@@ -28,60 +61,154 @@ contract DssSpellCollateralAction {
     // $ bc -l <<< 'scale=27; e( l(1.08)/(60 * 60 * 24 * 365) )'
     //
     // A table of rates can be found at
-    //    https://ipfs.io/ipfs/QmTRiQ3GqjCiRhh1ojzKzgScmSsiwQPLyjhgYSxZASQekj
+    //    https://ipfs.io/ipfs/QmX2QMoM1SZq2XMoTbMak8pZP86Y2icpgPAKDjQg4r4YHn
     //
 
+    uint256 constant ZERO_ZERO_FIVE_PCT_RATE = 1000000000015850933588756013;
+
+    // --- Math ---
+    uint256 public constant WAD = 10**18;
+    uint256 public constant RAY = 10**27;
+    uint256 public constant RAD = 10**45;
+
+    // -- RWA008 MIP21 components --
+    address constant RWA008                    = 0xb9737098b50d7c536b6416dAeB32879444F59fCA;
+    address constant MCD_JOIN_RWA008_A         = 0xF4D4184413d97C8C8a4f0437AbEa711bc5991a7e;
+    address constant RWA008_A_URN              = 0x593f2A49A7aA5e0c333edB4e3abB10654F68069c;
+    address constant RWA008_A_URN_CLOSE_HELPER = 0xCfc4043675EE82EEAe63C90D6eb3aB2dcf833431;
+    address constant RWA008_A_INPUT_CONDUIT    = 0x7032546Ba3F6E8866334556a354e67B905aA4470;
+    address constant RWA008_A_OUTPUT_CONDUIT   = 0x21CF5Ad1311788D762f9035829f81B9f54610F0C;
+    // SocGen's wallet
+    address constant RWA008_A_OPERATOR         = 0x03f1A14A5b31e2f1751b6db368451dFCEA5A0439;
+    // DIIS Group wallet
+    address constant RWA008_A_MATE             = 0xb9444802F0831A3EB9f90E24EFe5FfA20138d684;
+
+    string  constant RWA008_DOC                = "QmdfzY6p5EpkYMN8wcomF2a1GsJbhkPiRQVRYSPfS4NZtB";
+    /**
+     * The Future Value of the debt ceiling by the end of the agreement:
+     *   - 30,000,00 USD: Debt Ceiling
+     *   - 0.05% per year: Stability Fee
+     *   - 2.9 years: Duration of the Loan
+     *
+     *     bc -l <<< 'scale=18; (30000000 * e( l(1.0005) * 2.9 ))'
+     */
+    uint256 constant RWA008_A_INITIAL_PRICE    = 30_043_520_665599336150000000;
+    uint48  constant RWA008_A_TAU              = 0;
+
+    // Ilk registry params
+    uint256 constant RWA008_REG_CLASS_RWA      = 3;
+
+    // Remaining params
+    uint256 constant RWA008_A_LINE             = 30_000_000;
+    uint256 constant RWA008_A_MAT              = 100_00; // 100% in basis-points
+    uint256 constant RWA008_A_RATE             = ZERO_ZERO_FIVE_PCT_RATE;
+    // -- RWA008 end --
+
+    function onboardRwa008(
+        ChainlogAbstract CHANGELOG,
+        IlkRegistryAbstract REGISTRY,
+        address MIP21_LIQUIDATION_ORACLE,
+        address MCD_VAT,
+        address MCD_JUG,
+        address MCD_SPOT,
+        address MCD_JOIN_DAI
+    ) internal {
+        // RWA008-A collateral deploy
+        bytes32 ilk      = "RWA008-A";
+        uint256 decimals = DSTokenAbstract(RWA008).decimals();
```

Comment:
Why not just to use GemJoinAbstract and remove this dependency?

Reply:
Fixed.